src/daemon/XorgDisplayServer.cpp

```diff
@@ -233,6 +233,23 @@ namespace SDDM {
         // log message
         qDebug() << "Display server stopped.";
 
+        QString displayStopCommand = mainConfig.XDisplay.DisplayStopCommand.get();
+
+        // create display setup script process
+        QProcess *displayStopScript = new QProcess();
+
+        // set process environment
+        QProcessEnvironment env;
+        env.insert("DISPLAY", m_display);
+        env.insert("HOME", "/");
+        env.insert("PATH", mainConfig.Users.DefaultPath.get());
+        env.insert("SHELL", "/bin/sh");
+        displayStopScript->setProcessEnvironment(env);
+
+        // start display setup script
+        qDebug() << "Running display stop script " << displayStopCommand;
+        displayStopScript->start(displayStopCommand);
```

Comment:
On 02-04-15 08:56:57, davidedmundson wrote:
>    In [1]src/daemon/XorgDisplayServer.cpp:
> > +        QString displayStopCommand = mainConfig.XDisplay.DisplayStopCommand.g
> et();
> > +
> > +        // create display setup script process
> > +        QProcess *displayStopScript = new QProcess();
> > +
> > +        // set process environment
> > +        QProcessEnvironment env;
> > +        env.insert("DISPLAY", m_display);
> > +        env.insert("HOME", "/");
> > +        env.insert("PATH", mainConfig.Users.DefaultPath.get());
> > +        env.insert("SHELL", "/bin/sh");
> > +        displayStopScript->setProcessEnvironment(env);
> > +
> > +        // start display setup script
> > +        qDebug() << "Running display stop script " << displayStopCommand;
> > +        displayStopScript->start(displayStopCommand);
>
>    should we block for it to finish?

Yes, that would be a good idea. It's been a while since I last used Qt
and QProcess.

--
Alberto Milone